### ISSUE TITLE
[Navigation][Accessibility] Inconsistent isActive behavior

### ISSUE THREAD

**Issue by jakublamprecht** [adobe/aem-core-wcm-components#688]
## Bug Report

**Current Behavior**
In `NavigationItem.java` file there is a documentation for `isActive` method which says, that it returns true if the current page is represented by this item, and false otherwise:

```
    /**
     * Returns {@code true} if the page contained by this navigation item is active.
     *
     * @return {@code true} if it is the current page, otherwise {@code false}
     * @since com.adobe.cq.wcm.core.components.models 11.0.0; marked <code>default</code> in 12.1.0
     */
```

This is not exactly correct - `isActive` is true if the item represents current page or is a parent of the current page - this is leveraged in the markup of navigation to add the `cmp-navigation__item--active`.

This is a problem for `aria-current` attribute which should be set only on the link to the current page, but currently, because it uses said `isActive` flag, the `aria-current` attribute is present also on any parent of the item.

**Expected behavior/code**

Only the item that represents current page should have `aria-current="page"` attribute.

**Environment**
- Core Components 2.5.0

**Comment by msagolj** [comment]
@jakublamprecht you are right, the implementation is correct but the JavaDoc is wrong, it should say:

'if it is the current page or one of its ancestors'

 and the code for aria-current is using it incorrectly.

**Comment by msagolj** [comment]
(CQ-4277267)

**Comment by gabrielwalt** [comment]
We could just change the `aria-current="page"` attribute to be set only for the current page without requiring a new version of the component.
https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/navigation/v1/navigation/itemContent.html#L18
